lib/fun_with_flags/ui/router.ex

```diff
@@ -122,9 +122,14 @@ defmodule FunWithFlags.UI.Router do
   # to clear a boolean gate
   #
   delete "/flags/:name/boolean" do
-    flag_name = String.to_existing_atom(name)
-    FunWithFlags.clear(flag_name, boolean: true)
-    redirect_to conn, "/flags/#{name}"
+    case Utils.get_flag(name) do
```

Comment:
Unfortunately under the hood this still calls `String.to_existing_atom(name)`, so it just moves the problem somewhere else.

Reply:
`Utils.get_flag/1` seems to use `safe_flag_exists?/1`, which does make the operation safe, as far as I understand 😄

Reply:
Not really:

* https://github.com/tompave/fun_with_flags_ui/blob/3b0e4810456906fd3456c158a512087b0514b8f5/lib/fun_with_flags/ui/utils.ex#L154
* https://github.com/tompave/fun_with_flags_ui/blob/3b0e4810456906fd3456c158a512087b0514b8f5/lib/fun_with_flags/ui/utils.ex#L77

It makes it "safe" because it doesn't automatically create Atoms from user input. But it still blows up (by design) with Atoms that don't already exist in the VM.

Reply:
Right, sorry. I didn't see the `rescue` in `safe_flag_exists?/1` 🤦

Reply:
Yeah but the function also calls `FunWithFlags.all_flag_names` which in turn calls the persistence adapter `all_flag_names` function, which calls `String.to_atom/1` on all flags (as far as I understand, again)

https://github.com/tompave/fun_with_flags/blob/5a5fdeabc2262bf5a797c522a3bf24dbb67d2f6b/lib/fun_with_flags/store/persistent/redis.ex#L135

https://github.com/tompave/fun_with_flags/blob/5a5fdeabc2262bf5a797c522a3bf24dbb67d2f6b/lib/fun_with_flags/store/persistent/ecto.ex#L247